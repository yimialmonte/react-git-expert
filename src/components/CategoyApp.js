import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoyApp = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 1) {
      setCategories((prev) => [...prev, inputValue]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInput} />
    </form>
  );
};

CategoyApp.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default CategoyApp;
