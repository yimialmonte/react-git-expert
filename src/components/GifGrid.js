import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGif';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(setImages);
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((item) => (
          <GifGridItem key={item.id} {...item} />
        ))}
      </div>
    </>
  );
};
