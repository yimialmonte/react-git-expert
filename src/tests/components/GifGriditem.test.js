import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test on <GifGridItem />', () => {
  const title = 'Title';
  const url = 'https://localhost/logo.png';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Should show the correct component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should have a paragraph with title', () => {
    const paragraph = wrapper.find('p');
    expect(paragraph.text().trim()).toBe(title);
  });

  test('Should image display correctly', () => {
    const image = wrapper.find('img');
    expect(image.props().src).toBe(url);
    expect(image.props().alt).toBe(title);
  })

  test('Should have animate__fadeIn class', () => {
    const element = wrapper.find('div');
    const className = element.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  })

});
