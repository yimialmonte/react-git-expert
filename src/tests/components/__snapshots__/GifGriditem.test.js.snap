// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem /> Should show the correct component 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Title"
    src="https://localhost/logo.png"
  />
  <p>
    Title
  </p>
</div>
`;
